### DISCUSSION TITLE
Option to run some containers in VM

### DISCUSSION THREAD

**Discussion by caukub** [pelican-dev/panel#24] [Ideas]
I think having option to run some services in "real" isolation using VM for extra security could be beneficial to some users. This could be easily implemented with Docker (OCI) compatible runtime like KataContainers.

**Comment by ZEROPOINTBRUH** [comment]
Not sure how that would work since pterodactyl is completely docker based. Also not to mention people running a vm over a vm may cause major performance loss.

**Comment by jacksonm36** [comment]
it could utilize kubernetes? maybe

**Reply by ZEROPOINTBRUH** [reply]
That, can work.

**Comment by lancepioch** [comment]
We are interested in this, but to be honest, this is going to be very far down the road and at our convenience! I'm not kidding when I say this, but please check back in next year 😅
